main.py

```
@@ -236,6 +263,172 @@ async def api_compress_pdf(
+            temp_paths.append(temp_path)
+
+        pwd_list = None
+        if passwords:
+            pwd_list = [p if p else None for p in passwords.split(",")]
+
+        output_path = await run_in_threadpool(
+            merge_pdfs, [str(p) for p in temp_paths], str(OUTPUT_DIR), pwd_list
+        )
+        return {"status": "success", "message": "PDFs merged", "filename": Path(output_path).name}
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)